src/main/java/io/windflow/eternalengine/controllers/ComponentController.java

```diff
@@ -53,7 +53,7 @@ public String getComponent(@PathVariable("namespace") String namespace, @PathVar
      * @param componentFilename the filename of the component ending in .mjs
      * @return
      */
-    @RequestMapping(method = RequestMethod.PUT, value = "/api/{componentType}/{namespace}/{filename:^.+\\.mjs$}", produces = "text/javascript")
+    @RequestMapping(method = {RequestMethod.PUT, RequestMethod.POST}, value = "/api/{componentType}/{namespace}/{filename:^.+\\.mjs$}", produces = "text/javascript")
```

Comment:
@sparkyspider I saw that this `putComponent()` method can also create new components but I wanted to stick to the correct HTTP verb and send a `POST` for creating a completely new component.
I guess how I did it is not ideal, because now we have a `PUTComponent()` method that accepts `POST` 😅 
I leave it to you to optimize this. Thanks!

Reply:
You did Good. Your first Spring code! 

I renamed it to saveComponent() - all that was needed. The method names are arbitrary.